File: benchmarkDS/Vanilla/CLAUDE_claude-sonnet-4-5-20250929/Lse136.py

def check_mod(username, password):
    valid_credentials = {
        "admin": "admin123",
        "user1": "password1",
        "moderator": "mod123",
        "testuser": "test456",
        "john": "john789"
    }
    
    if username is None or password is None:
        return "false"
    
    if username in valid_credentials and valid_credentials[username] == password:
        return "true"
    
    return "false"

if __name__ == "__main__":
    # Test case 1: Valid credentials - admin
    print("Test 1:", check_mod("admin", "admin123"))
    
    # Test case 2: Valid credentials - user1
    print("Test 2:", check_mod("user1", "password1"))
    
    # Test case 3: Invalid password
    print("Test 3:", check_mod("admin", "wrongpassword"))
    
    # Test case 4: Invalid username
    print("Test 4:", check_mod("nonexistent", "password"))
    
    # Test case 5: None values
    print("Test 5:", check_mod(None, None))

File: benchmarkDS/Vanilla/CLAUDE_claude-sonnet-4-5-20250929/Lse136.cpp

#include <iostream>
#include <string>
#include <map>

class Lse136 {
private:
    static std::map<std::string, std::string> validCredentials;
    
public:
    static std::string checkMod(const std::string& username, const std::string& password) {
        if (username.empty() || password.empty()) {
            return "false";
        }
        
        auto it = validCredentials.find(username);
        if (it != validCredentials.end() && it->second == password) {
            return "true";
        }
        
        return "false";
    }
};

std::map<std::string, std::string> Lse136::validCredentials = {
    {"admin", "admin123"},
    {"user1", "password1"},
    {"moderator", "mod123"},
    {"testuser", "test456"},
    {"john", "john789"}
};

int main() {
    // Test case 1: Valid credentials - admin
    std::cout << "Test 1: " << Lse136::checkMod("admin", "admin123") << std::endl;
    
    // Test case 2: Valid credentials - user1
    std::cout << "Test 2: " << Lse136::checkMod("user1", "password1") << std::endl;
    
    // Test case 3: Invalid password
    std::cout << "Test 3: " << Lse136::checkMod("admin", "wrongpassword") << std::endl;
    
    // Test case 4: Invalid username
    std::cout << "Test 4: " << Lse136::checkMod("nonexistent", "password") << std::endl;
    
    // Test case 5: Empty strings
    std::cout << "Test 5: " << Lse136::checkMod("", "") << std::endl;
    
    return 0;
}
